### PR TITLE
Capture detailed Erlang logs on connection failure

### PR DESCRIPTION
**Pull request by dcorbacho**
Closes https://github.com/rabbitmq/rabbitmq-server/issues/401

### CODE REVIEW

File: src/rabbit_error_logger_handler.erl [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+code_change(_OldVsn, State, _Extra) ->
+    {ok, State}.
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+format({check_dflag_xnc_failed, _What}) ->
+    {"  * Remote node uses an incompatible Erlang version ~n", []};
+format({recv_challenge_failed, no_node, Node}) ->
+    {"  * Hostname mismatch ~p~n", [Node]};
```
Comment (michaelklishin) [inline src/rabbit_error_logger_handler.erl]: I feel we can improve this further. @dcorbacho this case is when remote node name as seen by the remote node itself differs from what the local node used, is my understanding correct? If so, what does the `Node` value indicate?

**Comment by michaelklishin** [comment]
I've updated the wording, @dcorbacho can you please verify that my understanding of each case is correct?

**Comment by michaelklishin** [comment]
Also, this depends on an OTP patch. Is it included in a released Erlang version?

File: src/rabbit_error_logger_handler.erl (hunk 1 above)
Comment (dcorbacho) [inline src/rabbit_error_logger_handler.erl]: `Node` is the remote node name as seen by the remote node itself. The name is received in the challenge exchange, and it is the local node that rejects it.

**Comment by dcorbacho** [comment]
@michaelklishin your wording is much better than mine, all correct.

**Comment by dcorbacho** [comment]
The OTP patch is not in any released Erlang version yet. It is merged into master, so it will be available in the next major release (estimated Q2 2016).

**Pull request merged**